app/src/main/java/be/humanoids/webthingify/Phone.java

```diff
@@ -60,11 +63,16 @@
                 }
             }
             if (cameraId != null) {
-                Map<String, Object> onDescription = new HashMap<>();
-                onDescription.put("@type", "OnOffProperty");
-                onDescription.put("label", "Flashlight On/Off");
-                onDescription.put("type", "boolean");
-                onDescription.put("description", "Whether the flashlight is turned on");
+                JSONObject onDescription = new JSONObject();
+                try {
+                    onDescription.put("@type", "OnOffProperty");
+                    onDescription.put("label", "Flashlight On/Off");
+                    onDescription.put("type", "boolean");
+                    onDescription.put("description", "Whether the flashlight is turned on");
+                } catch (JSONException e) {
+                    Log.e("wt:build", e.toString());
```

Comment:
Note: you can pass through an exception to `Log.e` as third param for nice logging of exceptions.

Reply:
Updated.